hs_core/views/resource_folder_hierarchy.py

```diff
@@ -172,7 +172,9 @@ def data_store_folder_unzip(request):
     try:
         unzip_file(user, res_id, zip_with_rel_path, bool_remove_original)
     except SessionException as ex:

```

Comment:
Given the time and other priorities on this project, the current solution may be "permanent" for a long time so it would be better if it were unit tests.  We rely on unit tests to validate that changes made elsewhere later on by others (there are many cooks in our kitchen) do not break existing functionality.  It is hard (impossible) for programmers to anticipate all the ramifications of code changes in other parts of the system, so these tests are a key defensive strategy, and the default should be "test" not "defer because it is interim and we may change later".  We should also not see solutions as "interim" to change later.  They should be "final" with the option to "enhance" later.  What we do now should be "complete" to the extent possible.